### PR TITLE
feat: Add configurable Helm version (#3872)

### LINKED ISSUE #3872: Configurable Helm Version
# Summary

Allow passing the version of helm to use for templating in the application crd / cli.

# Motivation

StatefulSets don't allow changing of labels. Helm has a `.Release.Service` value that is not over-ridable on the command line and this is used in most popular charts to set a label: `heritage: <value>`

I have some applications (rabbitmq, cert-manager) that are installed with helm (version3) and I am trying to adopt these intalls with ArgoCD. Unfortunately for rabbitmq chart, it's a stateful set and cannot change the `heritage` label.

Helm v3 cli sets this label with the value `Helm`, but Helm v2 uses the value `Tiller`. ArgoCD chooses the helm binary based on the chart.yaml `apiVersion` field. Most charts however will still use `v1` for backwards compatibility (I assume).

This means ArgoCD will use helm v2 instead of helm v3 to template most charts. As the value of `.Release.Service` is set by the version of helm being run, this means I need to be able to choose the version of helm that ArgoCD uses. Currently this is not possible and the alternatives are not attractive propositions:

- Hosting own versions of the charts which I'd need to pull and compare on each version increment.
- Asking chart maintainers to adopt `v2` in the `apiVersion` field as it will likely not change unless the chart has a feature dependant on helm3 as not to break backwards compatibility. 

See full discussion under #3735 

# Proposal

Add a new field to the Application CRD to allow setting the version of helm the install uses:

```yaml
spec:
  project: default

  source:
    repoURL: https://github.com/argoproj/argocd-example-apps.git
    targetRevision: HEAD
    path: guestbook

    helm:
      version: v3
      parameters:
      - name: "nginx-ingress.controller.service.annotations.external-dns\\.alpha\\.kubernetes\\.io/hostname"
        value: mydomain.example.com
      releaseName: guestbook
      valueFiles:
      - values-prod.yaml
      values: 
```

In the CLI there may be an optional parameter such as:

```
--helm-version string|int      Specify version of helm to template chart with  
```

If unspecified then the value should be set with the current logic of looking up the `apiVersion` field in chart.yaml.

This is unless a global preference could be specified across an instance of ArgoCD. My preference would be to default to the latest version of helm and only specify the version when `v2` is necessary. However this may be specific to my usage scenario. (Consider this part a stretch goal).

### PR DESCRIPTION
**Pull request by loxley**
This commit adds support for configurable Helm version either via
declarative syntax or via argocd cli.

New helm option 'Version' added to the ApplicationSourceHelm
struct which can be either 'v2' or 'v3'.

Argocd app create accepts '--helm-version' that also looks for the
same Helm versions as above.

Closes #3872 .

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Optional. My organization is added to USERS.md.
* [x] I've signed the CLA and my build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

### CODE REVIEW

File: pkg/apis/application/v1alpha1/types.go [hunk 1]
```diff
@@ -181,16 +181,18 @@ const (
 
 // ApplicationSourceHelm holds helm specific options
 type ApplicationSourceHelm struct {
+	// Version is the Helm version to use for templating with
+	Version string `json:"version,omitempty" protobuf:"bytes,1,opt,name=version"`
```
Comment (jannfis) [inline pkg/apis/application/v1alpha1/types.go:185]: For backwards compatibility, please make the new member the last field of protobuf (i.e. prepend, not append)

**Comment by jannfis** [comment]
Hey, this is cool! Will take a closer look tomorrow :)

**Comment by loxley** [comment]
> Hey, this is cool! Will take a closer look tomorrow :)

Cool, I think there is something with the e2e tests that I might have missed. I will also take a look at that tomorrow, could be I might need some help to get the correct data. :)

File: pkg/apis/application/v1alpha1/types.go (hunk 1 above)
Comment (loxley) [reply pkg/apis/application/v1alpha1/types.go:185]: Roger! Will fix!

Comment (loxley) [reply pkg/apis/application/v1alpha1/types.go]: Fixed in https://github.com/argoproj/argo-cd/commit/cbb5b270f72edc75ba8e61b8fe4fc6f0dae0ca3f

**Comment by loxley** [comment]
Commit https://github.com/argoproj/argo-cd/commit/cbb5b270f72edc75ba8e61b8fe4fc6f0dae0ca3f passed all 15 checks but after adding 3 new tests and pushing commit https://github.com/argoproj/argo-cd/commit/66994146547759e7c40f186489e7cae0ebaaf868 the e2e test failed.

**Review by jannfis — APPROVED** [review]
LGTM, @alexmt do you want to take an additional look please?

**Review by alexmt — APPROVED** [review]
LGTM!

**Comment by wrdls** [comment]
@alexmt Any idea when this will be included in a release? I checked the `1.7.5` tag but it doesn't seem the be included.

I checked https://github.com/argoproj/argo-cd/blob/v1.7.5/cmd/argocd/commands/app.go#L739 for example but `version` is not in `appOptions`.

**Pull request merged**